dash/development/component_generator.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import print_function
+
+import json
+import sys
+import subprocess
+import shlex
+import os
+
+from .component_loader import generate_imports
+from .base_component import generate_class_file
+
+
+# pylint: disable=too-many-locals
+def generate_components(component_src, project_shortname):
+    is_windows = sys.platform == 'win32'
+
+    extract_path = os.path.abspath(os.path.join(
+        os.path.dirname(__file__),
+        '..',
+        'extract-meta.js'
+    ))
+
+    os.environ['NODE_PATH'] = 'node_modules'
+    cmd = shlex.split('node {} {}'.format(extract_path, component_src),
+                      posix=not is_windows)
+
+    proc = subprocess.Popen(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=is_windows)
+    out, err = proc.communicate()
+    status = proc.poll()
+
+    if err:
+        print(err.decode(), file=sys.stderr)
+
+    if not out:
+        print(
+            'Error generating metadata in {} (status={})'.format(
+                project_shortname, status),
+            file=sys.stderr)
+        sys.exit(1)
+
+    metadata = json.loads(out.decode())
+
+    components = []
+
+    for component_path, component_data in metadata.items():
```

Comment:
as a general comment, this seems to be repeated code with https://github.com/plotly/dash/blob/d892b9e8d1c5a617eb1d9e5c2a23ea47bce558c2/dash/development/component_loader.py#L68 ... any way to DRY it up?